lib/modules/datasource/rubygems/get.ts

```diff
@@ -182,4 +218,110 @@ export class InternalRubyGemsDatasource extends Datasource {
+  } {
+    const endPointRe = regEx(
+      `^(?<nexusRegistry>.*:\\d{1,5}).*\\/repository\\/(?<repository>\\w+)\\/?`
+    );
+    const { nexusRegistry, repository } = url.match(endPointRe)?.groups ?? {};
+    return { nexusRegistry, repository };
+  }
+
+  @cache({
+    namespace: 'datasource-nexus',
+    key: (nexusEndPoint: string) => `isNexusDataSource:${nexusEndPoint}`,
+    ttlMinutes: 1440,
+  })
+  async isNexusDataSource(nexusEndPoint: string): Promise<boolean> {
+    const statusEndPoint = '/service/rest/v1/status';
+    const nexusUrl = joinUrlParts(nexusEndPoint, statusEndPoint);
+    try {
+      const response = await this.http.getJson<HttpResponse>(nexusUrl);
+      return this.extractServerFrom(response).startsWith('Nexus/') ?? false;
+    } catch (err) {
+      // istanbul ignore if
+      if (err.response?.headers?.server?.startsWith('Nexus/')) {
+        this.handleGenericErrors(err);
+      }
+    }
+    return false;
+  }
+
+  private extractServerFrom(
+    response: HttpResponse<HttpResponse<string>>
+  ): string {
+    let headerRespose = '';
+    // istanbul ignore else: not testable with nock
+    if (is.string(response.headers?.server)) {
+      headerRespose = response.headers?.server;
+    } else if (is.array(response.headers?.server)) {
+      [headerRespose] = response.headers?.server ?? [''];
+    }
+    return headerRespose;
+  }
+
+  private async getReleasesFromNexus(
+    repository: string,
+    nexusRegistry: string,
+    dependencyName: string
+  ): Promise<Release[]> {
+    const result: Release[] = [];
+    const gemItems: NexusGemsItems[] = [];
+    const url = this.getNexusGemReleasesEndPoint(
+      repository,
+      nexusRegistry,
+      dependencyName
+    );
+    const nexusResponse = (await this.http.getJson<NexusGems>(url))?.body || {
+      continuationToken: null,
+      items: [],
+    };
+    gemItems.push(...nexusResponse.items);
+    const continuationToken: string | null = nexusResponse.continuationToken;
+    if (continuationToken) {
+      const paginationGems = await this.paginationResult(
+        url,
+        continuationToken
+      );
+      gemItems.push(...paginationGems);
+    }
+
+    result.push(
+      ...gemItems.map((item) => {
+        return { version: item.version, rubyPlatform: 'nexus' };
+      })
+    );
+    return result;
+  }
+
+  private getNexusGemReleasesEndPoint(
+    repository: string,
+    nexusRegistry: string,
+    depName: string
+  ): string {
+    const nexusSearchPoint = '/service/rest/v1/search';
+    const query = `${nexusSearchPoint}?repository=${repository}&name=${depName}&sort=version`;
+    return joinUrlParts(nexusRegistry, query);
+  }
+
+  private async paginationResult(
+    url: string,
+    continuationToken: string
+  ): Promise<NexusGemsItems[]> {
+    const gemItems: NexusGemsItems[] = [];
+    let pageToken: string | null = continuationToken;
+    let pageCounter = 0;
+    while (pageToken && pageCounter < 1000) {
```

Comment:
that are a lot requests, is there a page size parameter we can raise to have less requests?

Reply:
Ah yes thought about that and they also mention it in their documentation but oddly with no reference to it.

![image](https://user-images.githubusercontent.com/90554456/177280216-fa60d1a4-b1c7-4b97-b297-e6520e004cb8.png)

I looked it up and there is a user who asked on stackoverflow why and sadly it does not seem to be settable by the request URL: https://stackoverflow.com/questions/52181412/how-to-set-page-size-on-sonatype-nexus3-api
It seems to be set to 50. Lets decrease the count limit ?

Reply:
i think 100 pages would be enough, that are 5000 releases.

@rarkins WDYT?

Reply:
Yes, let's leave it as that for now